Appraisals

```diff
@@ -24,3 +24,14 @@ appraise 'rails-6.1' do
   # Taken from https://github.com/rails/rails/pull/42366
   gem 'net-smtp', require: false
 end
+
+appraise 'rails-7.0' do
+  gem 'railties', '~> 7.0.1' # Rails 7.0.0 and Ruby 3.1 are incompatible. See https://github.com/rails/rails/issues/43998 and https://github.com/rails/rails/pull/43951
+  gem 'devise', '>= 4.8.1' # devise introduced Rails 7 support
+  gem 'minitest-rails', '~> 7.0.0'
+
+  # net-smtp, net-imap and net-pop were removed from default gems in Ruby 3.1, but is used by the `mail` gem.
+  # So we need to add them as dependencies until `mail` is fixed: https://github.com/mikel/mail/pull/1439
```

Comment:
Totally pedantic note:  the referenced PR was closed in favor of https://github.com/mikel/mail/pull/1472 and mail [2.8.0.rc1](https://rubygems.org/gems/mail/versions/2.8.0.rc1) fixes this, but, alas, it is still a RC.  There's been a request to back port it to mail 2.7, but that hasn't happened.

But I would have done exactly what you did here.

Reply:
Yes, in this case I just replicated the comment that was for the other versions, but as soon as it is a stable release I will open a PR to adjust the minimum version of the mail.